[AWS Block Support] Invalid memory address or nil pointer dereference for AWS Volume Creation

**Issue/Feature Description:** AWS Block Creation is failing with the following reason:

Invalid memory address or nil pointer dereference

![image](https://user-images.githubusercontent.com/8712951/94398421-25b69b80-0183-11eb-8ef8-9ef62fe82f92.png)

Error logs:
```
[2020-09-28T12:00:41+05:30] [ERROR] [/root/gopath/src/github.com/opensds/multi-cloud/block/pkg/service/service.go] [service.(*blockService).CreateVolume():166] [PID:1] Received error in creating volume at backend , error: Get object from backend failed.
2020-09-28 12:00:41  file=grpc.go:372 level=error panic recovered: runtime error: invalid memory address or nil pointer dereference
2020-09-28 12:00:41  file=grpc.go:373 level=error goroutine 17513 [running]:
runtime/debug.Stack(0xc00000c4e0, 0xc0009fc002, 0xc0009fc0c0)
        /usr/local/go/src/runtime/debug/stack.go:24 +0x9d
github.com/micro/go-micro/v2/server/grpc.(*grpcServer).processRequest.func1.1(0xc00076b250)
        /root/gopath/pkg/mod/github.com/micro/go-micro/v2@v2.6.0/server/grpc/grpc.go:373 +0x2cf
panic(0x1838480, 0x2c43360)
        /usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/opensds/multi-cloud/block/pkg/service.(*blockService).CreateVolume(0xc0005651c0, 0x1ef09c0, 0xc00095cc30, 0xc00095c030, 0xc00095cae0, 0x498f1c, 0x1931f40)
        /root/gopath/src/github.com/opensds/multi-cloud/block/pkg/service/service.go:167 +0x387
github.com/opensds/multi-cloud/block/proto.(*blockHandler).CreateVolume(0xc0007d10e0, 0x1ef09c0, 0xc00095cc30, 0xc00095c030, 0xc00095cae0, 0x7f5223ccc460, 0x0)
        /root/gopath/src/github.com/opensds/multi-cloud/block/proto/block.pb.micro.go:153 +0x5b
reflect.Value.call(0xc0007d2480, 0xc0008302b8, 0x13, 0x1b16bcc, 0x4, 0xc00013f1b0, 0x4, 0x4, 0x16, 0x192d5c0, ...)
        /usr/local/go/src/reflect/value.go:460 +0x5f6
reflect.Value.Call(0xc0007d2480, 0xc0008302b8, 0x13, 0xc00013f1b0, 0x4, 0x4, 0x203000, 0xc00013f1b0, 0x40c1d6)
        /usr/local/go/src/reflect/value.go:321 +0xb4
github.com/micro/go-micro/v2/server/grpc.(*grpcServer).processRequest.func1(0x1ef09c0, 0xc00095cc30, 0x1f06e20, 0xc0007f9c00, 0x1931f40, 0xc00095cae0, 0x0, 0x0)
        /root/gopath/pkg/mod/github.com/micro/go-micro/v2@v2.6.0/server/grpc/grpc.go:378 +0x303
github.com/micro/go-micro/v2/util/wrapper.AuthHandler.func1.1(0x1ef09c0, 0xc00095cbd0, 0x1f06e20, 0xc0007f9c00, 0x1931f40, 0xc00095cae0, 0xc00095cbd0, 0xc0000e2620)
        /root/gopath/pkg/mod/github.com/micro/go-micro/v2@v2.6.0/util/wrapper/wrapper.go:282 +0x5e3
github.com/micro/go-micro/v2/util/wrapper.TraceHandler.func1.1(0x1ef0900, 0xc000840f80, 0x1f06e20, 0xc0007f9c00, 0x1931f40, 0xc00095cae0, 0x16, 0x1931f40)
        /root/gopath/pkg/mod/github.com/micro/go-micro/v2@v2.6.0/util/wrapper/wrapper.go:122 +0x1c6
github.com/micro/go-micro/v2/util/wrapper.HandlerStats.func1.1(0x1ef0900, 0xc000840f80, 0x1f06e20, 0xc0007f9c00, 0x1931f40, 0xc00095cae0, 0x12, 0x0)
        /root/gopath/pkg/mod/github.com/micro/go-micro/v2@v2.6.0/util/wrapper/wrapper.go:67 +0x83
github.com/micro/go-micro/v2/server/grpc.(*grpcServer).processRequest(0xc0001ef380, 0x1efc7a0, 0xc0005ba000, 0xc0008455c0, 0xc0007e6120, 0xc0004c4100, 0x16, 0x1ef0900, 0xc000840f80, 0x203000, ...)
        /root/gopath/pkg/mod/github.com/micro/go-micro/v2@v2.6.0/server/grpc/grpc.go:395 +0x7ee
github.com/micro/go-micro/v2/server/grpc.(*grpcServer).handler(0xc0001ef380, 0x0, 0x0, 0x1efc7a0, 0xc0005ba000, 0x0, 0x0)
        /root/gopath/pkg/mod/github.com/micro/go-micro/v2@v2.6.0/server/grpc/grpc.go:314 +0x77f
google.golang.org/grpc.(*Server).processStreamingRPC(0xc000814300, 0x1f06e80, 0xc0005bc600, 0xc000152200, 0x0, 0xc0005650e0, 0x0, 0x0, 0x0)
        /root/gopath/pkg/mod/google.golang.org/grpc@v1.26.0/server.go:1237 +0xb42
google.golang.org/grpc.(*Server).handleStream(0xc000814300, 0x1f06e80, 0xc0005bc600, 0xc000152200, 0x0)
        /root/gopath/pkg/mod/google.golang.org/grpc@v1.26.0/server.go:1323 +0xc0a
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc0000426a0, 0xc000814300, 0x1f06e80, 0xc0005bc600, 0xc000152200)
        /root/gopath/pkg/mod/google.golang.org/grpc@v1.26.0/server.go:722 +0xbb
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /root/gopath/pkg/mod/google.golang.org/grpc@v1.26.0/server.go:720 +0xa1
```
**Why this issue to fixed / feature is needed(give scenarios or use cases):** To Support AWS Volume Creation with proper Error Handling

**How to reproduce, in case of a bug:**

1. Create AWS Backend for block service
2. Create AWS with 0 Bytes Size

**Other Notes /  Environment Information: (Please give the env information, log link or any useful information for this issue)**

--- a/block/pkg/service/service.go
+++ b/block/pkg/service/service.go
@@ -163,7 +163,10 @@ func (b *blockService) CreateVolume(ctx context.Context, in *pb.CreateVolumeRequ
 
 	vol, err := sd.CreateVolume(ctx, in)
 	if err != nil {
-		log.Errorf("Received error in creating volume at backend , error: %s ", err)
+		log.Errorf("received error in creating volume at backend , error: %s ", err)
+		if vol == nil {
+			return err
+		}
 		vol.Volume.Status = utils.VolumeStateError
 	} else {
 		vol.Volume.Status = utils.VolumeStateCreating
@@ -282,7 +285,10 @@ func (b *blockService) UpdateVolume(ctx context.Context, in *pb.UpdateVolumeRequ
 
 	vol, err := sd.UpdateVolume(ctx, in)
 	if err != nil {
-		log.Errorf("Received error in updating volumes at backend, error: %s ", err)
+		log.Errorf("received error in updating volumes at backend, error: %s ", err)
+		if vol == nil {
+			return err
+		}
 		vol.Volume.Status = utils.VolumeStateError
 	} else {
 		vol.Volume.Status = utils.VolumeStateUpdating

